app/controllers/users_controller.rb

```diff
@@ -0,0 +1,20 @@
+class UsersController < ApplicationController
+  def new
```

Comment:
Since you require login in ApplicationController here, you have to be logged in in order to register :-) that is probably not possible, as you can not log in before you register right.